aws-api/src/main/java/com/amplifyframework/api/aws/HeaderSupplier.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.api.aws;
+
+/**
+ * Provide an implementation of this HeaderSupplier to the AWS API plugin's builder.
+ * This supplier can be used to register customer headers with the API plugin.
+ * The supplier will be called whenever an outbound request is made.
+ * The plugin needs additional information in order for this simple functional
+ * interface to be usable. Those internal concerns are maintained in
+ * an internal state bundle, {@link HeaderSupplierInfo}.
+ */
+@FunctionalInterface
+public interface HeaderSupplier {
+    /**
+     * Supplies the next header value.
+     * @return A header value
+     */
+    String nextValue();
```

Comment:
At first I wasn't sure about the `nextValue()` name in the context of supplier (in my mind, when I see supplier I thing `get()`). But I think in the context of this change, `nextValue()` is OK since this method will be called for each request issued. Just wanted to confirm that this is in fact your interpretation.

Reply:
I updated the PR, and this method no longer exists.